content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md

```diff
@@ -796,4 +796,90 @@ webhooks:
   ...
 ```
 
+## Best practices and warnings
+
+### Idempotence
+
+An idempotent mutating admission webhook is able to successfully process an object it has already admitted
+and potentially modified. The admission can be applied multiple times without changing the result beyond
+the initial application.
+
+#### Example of idempotent mutating admission webhooks:
+
+1. For a `CREATE` pod request, set the field `.spec.securityContext.runAsNonRoot` of the
+   pod to true, to enforce security best practices.
+
+2. For a `CREATE` pod request, if the field `.spec.containers[].resouces.limits`
+   of a container is not set, set default resource limits.
+
+3. For a `CREATE` pod request, inject a sidecar container with name `foo-sidecar`.
+
+In the cases above, the webhook can be safely reinvoked, or admit an object that already has the fields set.
+
+#### Example of non-idempotent mutating admission webhooks:
+
+1. For a `CREATE` pod request, inject a sidecar container with name `foo-sidecar`
+   suffixed with the current timestamp (e.g. `foo-sidecar-19700101-000000`).
```

Comment:
add examples of a webhook failing on its own output:
* a webhook that rejects pods with the label "env" set, then adds an `env: prod` label
* a webhook that blindly appends a container named `foo-sidecar` without looking to see if there is already a `foo-sidecar` container in the pod